experimental/Schemes/BlowupMorphism.jl

```diff
@@ -892,3 +893,71 @@ function blow_up(m::AbsCoveredScheme, I::AbsIdealSheaf; coordinate_name = "e")
   @assert m === scheme(I) "Incompatible scheme and ideal sheaf"
   return blow_up(I)
 end
+
+########################################################################
+# strict transforms of ideal sheaves                                   #
+########################################################################
+
+@attributes mutable struct StrictTransformIdealSheaf{SpaceType, OpenType, OutputType,
+                                                     RestrictionType
+                                                    } <: AbsIdealSheaf{
+                                                                       SpaceType, OpenType,
+                                                                       OutputType, RestrictionType
+                                                                      }
+  morphism::AbsSimpleBlowdownMorphism
+  orig::AbsIdealSheaf
+  underlying_presheaf::AbsPreSheaf
+
+  function StrictTransformIdealSheaf(
+      f::AbsSimpleBlowdownMorphism,
+      J::AbsIdealSheaf
+    )
+    @assert scheme(J) === codomain(f)
+    X = domain(f)
+    Ipre = PreSheafOnScheme(X,
+                      OpenType=AbsAffineScheme, OutputType=Ideal,
+                      RestrictionType=Map,
+                      is_open_func=_is_open_func_for_schemes_without_affine_scheme_open_subscheme(X)
+                     )
+    I = new{typeof(X), AbsAffineScheme, Ideal, Map}(f, J, Ipre)
+    return I
+  end
+end
+
+morphism(I::StrictTransformIdealSheaf) = I.morphism
+original_ideal_sheaf(I::StrictTransformIdealSheaf) = I.orig
+underlying_presheaf(I::StrictTransformIdealSheaf) = I.underlying_presheaf
+
+function produce_object_on_affine_chart(I::StrictTransformIdealSheaf, U::AbsAffineScheme)
+  f = morphism(I)
+  X = domain(f)
+  Y = codomain(f)
+  J = original_ideal_sheaf(I)
+  @assert any(x->x===U, affine_charts(X))
+  E = exceptional_divisor(f)
+  IE = ideal_sheaf(E)
+  # We assume that the covering morphism has the default_covering of X as its domain.
```

Comment:
how restrictive is this assumption? will it explode otherwise?
In this please raise a useful error message.

Reply:
I don't see how this is resolved :P. But I can live with it.

Reply:
I think I added an assertion just the line below. Didn't I?

Reply:
right. I overlooked this. Sorry.